src/classes/RD2_ApiService.cls

```diff
@@ -145,18 +145,79 @@ public inherited sharing class RD2_ApiService {
             statusReason = rdSchedule.StatusReason__c;
             createdDate = rdSchedule.CreatedDate;
             lastModifiedDate = rdSchedule.LastModifiedDate;
+            currencyCode = cCode;
+        }
+    }
+
+    private String getCurrencyCode(npe03__Recurring_Donation__c rd) {
+        String currencyCode;
+
+        try {
+            if (UserInfo.isMultiCurrencyOrganization()) {
+                currencyCode = (String) rd.get(UTIL_Currency.CURRENCY_ISO_CODE_FIELD);
+            } else {
+                currencyCode = Userinfo.getDefaultCurrency();
+            }
+        } catch (Exception ex) {
+            // ignore any errors if for some reason we can't retrieve the currency code
+            // because usage in the ActiveSchedule inner class is only for api use
+        }
+
+        return currencyCode;
+    }
+
+    /**
+     * @description Retrieve a map of scheduled installments by Rd.id.
+     * @param rdIds Recurring Donation Ids to retrieve schedules for
+     * @return JSON Serialized version of Map<Id, List<RD2_ApiService.Installment>> installmentsByRdId
+     */
+    public String queryInstallments(Date startDate, Date endDate, Set<Id> rdIds) {
+        Map<Id, List<RD2_ApiService.Installment>> installmentsByRdId = new Map<Id, List<RD2_ApiService.Installment>>();
+
+        if (!RD2_EnablementService.isRecurringDonations2Enabled) {
+            return JSON.serialize(installmentsByRdId);
+        }
+
+        RD2_ScheduleService service = new RD2_ScheduleService();
+        String soql = getRecurringDonationQuery();
+        for (npe03__Recurring_Donation__c rd : Database.query(soql)) {
 
-            try {
-                if (UserInfo.isMultiCurrencyOrganization()) {
-                    currencyCode = (String) rd.get(UTIL_Currency.CURRENCY_ISO_CODE_FIELD);
-                } else {
-                    currencyCode = Userinfo.getDefaultCurrency();
-                }
-            } catch (Exception ex) {
-                // ignore any errors if for some reason we can't retrieve the currency code
-                // because usage in the ActiveSchedule inner class is only for api use
+            List<RD2_ApiService.Installment> installments = new List<RD2_ApiService.Installment>();
+            for (RD2_ScheduleService.Installment installment : service.getVisualizedInstallments(
```

Comment:
I suspect that the logic in the `getVisualizedInstallments()` will default the startDate and also exclude paused installments, but it might be worth adding something here so it's clear that is the behavior. Could be just something to set the startDate to the current date if null and whatever is needed to excluded paused installments.

Reply:
That behavior is a part of getVisualInstallments().  This is the version used calculate yearly totals.  Start Date default to currentDate is a behavior of RD2_ScheduleService.

Are you suggesting I document that or that I don't rely on it and provide my own implementation?

Reply:
No. However, reading this new method in the API I'm unclear on the start date and whether skipped installments are included or not.